File: Tests/src/pages/DepositPage.java
package pages;

import java.text.ParseException;
import java.util.List;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

import utilities.Dates;

public class DepositPage extends BasePage {
	@FindBy(id = "amount")
	private WebElement amount;
	
	@FindBy(id = "currency")
	private WebElement currency;
	
	@FindBy(id = "percent")
	private WebElement percent;
	
	@FindBy(id = "term")
	private WebElement term;
	
	@FindBy(xpath = "//td[text() = '360 days']/input")
	private WebElement finYear360;
	
	@FindBy(xpath = "//td[text() = '365 days']/input")
	private WebElement finYear365;
	
	@FindBy(id = "day")
	private WebElement day;
	
	@FindBy(id = "month")
	private WebElement month;
	
	@FindBy(id = "year")
	private WebElement year;
	
	@FindBy(id = "calculateBtn")
	private WebElement calculateBtn;
	
	@FindBy(id = "endDate")
	private WebElement endDate;
	
	@FindBy(id = "interest")
	private WebElement interest;
	
	@FindBy(id = "income")
	private WebElement income;
	
	@FindBy(xpath = "//div[text() = 'History']")
	private WebElement history;
	
	@FindBy(xpath = "//div[text() = 'Settings']")
	private WebElement settings;
	
	public DepositPage(WebDriver driver) {
		super(driver);
		PageFactory.initElements(this.driver, this);
	}

	public String getCurrency() {
		return currency.getText();
	}
	
	public String getFinYear() {
		if (finYear360.isSelected()) return "360";
		if (finYear365.isSelected()) return "365";
		
		return "";
	}

	public void setFinYear(String value) {
		switch (value) {
		case "360":
			finYear360.click();
			break;

		case "365":
			finYear365.click();
			break;
		}
	}
	
	public String getStartDate() {
		return getDay() + "/" + getMonth() + "/" + getYear();
	}
	
	public void setStartDate(String value) throws ParseException {
		String yyyy = Dates.formatDate(value, "yyyy");
		String mmmm = Dates.formatDate(value, "MMMM");
		String d = Dates.formatDate(value, "d");
		
		setYear(yyyy);
		setMonth(mmmm);
		setDay(d);
	}
	
	public String getDay() {
		String d = getSelectValue(day);
		String dd = String.format("%02d", Integer.parseInt(d));
		return dd;
	}
	public void setDay(String value) { setSelectValue(day, value); }
	public List<String> getDays(){ return getSelectOptions(day); }

	public String getMonth() {
		String mmmm = getSelectValue(month);
		String mm = String.format("%02d", mmmm);
		return mm;
	}
	public void setMonth(String value) { setSelectValue(month, value); }
	public List<String> getMonths(){ return getSelectOptions(month); }
	
	public String getYear() { return getSelectValue(year); }
	public void setYear(String value) { setSelectValue(year, value); }
	public List<String> getYears(){ return getSelectOptions(year); }

	public String getAmount() { return getInputValue(amount); }
	public void setAmount(String value) { setInputValue(amount, value); }
	
	public String getPercent() { return getInputValue(percent); }
	public void setPercent(String value) { setInputValue(percent, value); }

	public String getTerm() { return getInputValue(term); }
	public void setTerm(String value) { setInputValue(term, value); }

	public void populate(String amount, String percent, String term) {
		setAmount(amount);
		setPercent(percent);
		setTerm(term);
	}
	
	public void calculate() {
		calculateBtn.click();
	}

	public String getEndDate()  { return getInputValue(endDate); }
	public String getInterest() { return getInputValue(interest); }
	public String getIncome()   { return getInputValue(income); }
	
	public SettingsPage openSettings() {
		settings.click();
		return new SettingsPage(driver);
	}
}

File: Tests/pages/DepositPage.py
import calendar
from datetime import datetime
from time import sleep

from selenium.webdriver.common.by import By

from pages.BasePage import BasePage
from utils.Formatter import Formatter


class DepositPage(BasePage):
    def __init__(self, driver):
        super().__init__(driver)
        sleep(1)

    def open_settings(self):
        from pages.SettingsPage import SettingsPage
        self.find_element(By.XPATH, "//div[text() = 'Settings']").click()
        return SettingsPage(self.driver)

    def open_history(self):
        from pages.HistoryPage import HistoryPage
        self.find_element(By.XPATH, "//div[text() = 'History']").click()
        return HistoryPage(self.driver)

    def get_currency(self): return self.find_element(By.ID, "currency").text
    def get_amount(self): return self.get_input_value("amount")
    def set_amount(self, amount): self.set_input_value(amount, "amount")

    def get_percent(self): return self.get_input_value("percent")
    def set_percent(self, percent): self.set_input_value(percent, "percent")

    def get_term(self): return self.get_input_value("term")
    def set_term(self, term): self.set_input_value(term, "term")

    def get_fin_year(self):
        return "365" if self.find_element(By.XPATH, "//td[text() = '365 days']/input").is_selected() else "360"

    def set_fin_year(self, fin_year):
        if not(fin_year in ["360", "365"]):
            raise ValueError("Only 360 and 365 values are allowed")

        self.find_element(By.XPATH, "//td[text() = '" + fin_year + " days']/input").click()

    def get_start_date_days(self): return self.get_select_options("day")
    def get_start_date_months(self): return self.get_select_options("month")
    def get_start_date_years(self): return self.get_select_options("year")

    def set_start_date_day(self, value): self.select_in_dropdown("day", value)
    def set_start_date_month(self, value): self.select_in_dropdown("month", value)
    def set_start_date_year(self, value): self.select_in_dropdown("year", value)

    def set_start_date(self, date):
        self.select_in_dropdown("year", str(date.year))
        self.select_in_dropdown("month", calendar.month_name[date.month])
        self.select_in_dropdown("day", str(date.day))

    def get_start_date(self):
        d = self.get_selected_option("day")
        m = list(calendar.month_name).index(self.get_selected_option("month"))
        y = self.get_selected_option("year")

        return datetime(int(y), m, int(d)).date()

    def get_end_date(self): return self.get_input_value("endDate")
    def get_interest(self): return self.get_input_value("interest")
    def get_income(self): return self.get_input_value("income")

    def calculate(self, amount, percent, term, fin_year="365"):
        self.set_amount(amount)
        self.set_percent(percent)
        self.set_term(term)
        self.set_fin_year(fin_year)
        self.find_element(By.ID, "calculateBtn").click()
        self.wait_for_clickable(3, (By.ID, "calculateBtn"))

        return self

    def get_data(self, date_format, number_format):
        return [
            Formatter.format_number(self.get_amount(), number_format),
            self.get_percent() + "%",
            self.get_term(),
            self.get_fin_year(),
            Formatter.format_date(self.get_start_date(), date_format),
            self.get_end_date(),
            self.get_interest(),
            self.get_income()
        ]
